File: leecode/code/560/560.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int[] sum = new int[nums.length + 1];
        int result = 0;
        sum[0] = 0;
        for (int i = 1; i <= nums.length; i++) {
            sum[i] = sum[i - 1] +  nums[i - 1];
        }
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j <= nums.length; j++) {
                if (sum[j] - sum[i] == k) {
                    result++;
                }
            }
        }
        return result;
    }
}

File: leecode/code/560/560.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        record = {}
        record[0] = 1;
        res = 0
        pre = 0


        for i in nums:
            pre += i
            res += record.get(pre - k, 0)
            record[pre] = record.get(pre, 0) + 1


        return res 
            
            


